Model/Magento/CatalogImportExport/ResourceModel/Import/Data.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace FireGento\ExtendedImport\Model\Magento\CatalogImportExport\ResourceModel\Import;
+
+use FireGento\ExtendedImport\Model\Magento\CatalogImportExport\Import\Product\Product;
+
+/**
+ * Class Data
+ * @package FireGento\ExtendedImport\Model\Magento\CatalogImportExport\ResourceModel\Import
+ */
+class Data extends \Magento\ImportExport\Model\ResourceModel\Import\Data
+{
+    /**
+     * @param string $entity
+     * @param string $behavior
+     * @param array $data
+     * @return int
+     */
+    public function saveBunch($entity, $behavior, array $data)
+    {
+        $data = $this->overrideTheUrlKeys($data);
+
+        return parent::saveBunch($entity, $behavior, $data);
+    }
+
+    /**
+     * @param $rows
+     * @return mixed
+     */
+    private function overrideTheUrlKeys($rows)
+    {
+        $urlKeys = array_column($rows, Product::URL_KEY);
+        $occurrences = array_count_values($urlKeys);
+        $occurrencesIndex = [];
+
+        foreach ($urlKeys as $rowIndex => $urlKey) {
```

Comment:
does this really work? I think if a product A and B have the same name / url key I guess it is working. But only if A and B are imported in the same batch.
If A is already in the database, and B is imported now, would it work?

Instead of increasing occurences I would also be okay to use something like the product's  SKU ?

Reply:
@amenk Actually... I think it's much better to use SKU here, cause they are unique and the order / being imported or not doesn't matter. Will change that to SKU now, and it will cover all the cases.

Reply:
Actually you need to use the URL-sanitized SKU. which causes the problem, that if two SKUs result in the same URL part, again we have to use -1 -2 ...
SKU A = foobar 
SKU B = fòòbar
Product names the same
URL A = name-foobar
URL B = name-foobar

hmm...

Reply:
@amenk So.. How can we resolve that then?

Reply:
@avstudnitz what do you suggest?